app/views/users/index.html.erb

```diff
@@ -1,12 +1,26 @@
 <section class="users-section">
-  <ul class="users-list">
-    <% @users.each do |user| %>
-      <li>
-        Name: <%= user.name %>
-        <span class="profile-link">
-          <%= link_to 'See Profile',  user_path(user), class: 'profile-link' %>
-        </span>
-      </li>
-    <% end %>
-  </ul>
+  <%= render 'partials/friends'%>
+  <%= render 'partials/pending_requests' %>
+  <%= render 'partials/received_requests' %>
+
+  <div class="other_user">
+    <h3>Other Users</h3>
+    <ul class="users-list">
+      <% @users.each do |user| %>
+        <% unless user.name == current_user.name %>
+          <% if current_user.viable_friend?(user) %> 
+            <li>
+              <%= user.name %>
+              <span class="profile-link">
+                <%= link_to 'See Profile',  user_path(user), class: 'profile-link' %>              
+                  <%= link_to user_friendships_path(user), method: :post, data: {confirm: "Are you sure you want to add this friend?" } do %>
+                    <button>Add Friend</button>
+                  <% end %>
+              </span>
+            </li>
+          <% end %>
+        <% end %>
```

Comment:
- [ ] create a helper method that contains the unless and if statements calling a partial containing the markup within the if statement and then replace lines 10 to 22 with the helper method call instead of having the complexity straight in the view.